### PR TITLE
Miscellaneous minor updates to WinCompat

### LINKED ISSUE #11903: The remote session with the name WinPSCompatSession  is not available
# Steps to reproduce

```powershell
Import-Module ModuleName -UseWindowsPowerShell
```

# Expected behavior

```none
Module loads without errors.
```

# Actual behavior

```none
The remote session with the name WinPSCompatSession  is not available.
```

# Environment data

```none
PSVersion                      7.0.0-rc.2
PSEdition                      Core
GitCommitId                    7.0.0-rc.2
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by anmenaga**
## PR Summary

Several small fixes to some WinCompat scenarios:

1. Removing PS-Core-specific paths from `PSModulePath` of WinCompat process (Windows PS). Similar change was done previously by #11057 for NativeCommandProcessor. This PR enables same 
`PSModulePath` behavior for WinCompat process (Windows PS) and job's process (Windows PS - for jobs started with `-PSVersion 5.1`). Without this change there could be bad cases when WinCompat process was loading PS-Core versions of the modules.

1. Some implicit WinCompat scenarios were ignoring `-NoClobber` and `-Scope` parameters. This is now fixed.

1. `Get-PSSession -Name` generates an error when the session is not found (in addition to returning nothing). When WinCompat uses `Get-PSSession` to search for existing `WinPSCompatSession` session to reuse, such error is not shown to the user but it is written to the error stream. Considering that it is normal for `WinPSCompatSession` session to not be there before the first WinCompat module import, such error is expected but it may confuse user if it is found in the error stream. The fix is to do `Get-PSSession -Name WinPSCompatSession -ErrorAction Ignore`.
Fix #11903

1. Fixed a test bug in `ModulePath.Tests.ps1`.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs [hunk 1]
```diff
@@ -2359,7 +2359,7 @@ private IEnumerable<PSModuleInfo> CreateFakeModuleObject(IEnumerable<ModuleSpeci
                 {
                     if (importingModule)
                     {
-                        IList<PSModuleInfo> moduleProxies = ImportModulesUsingWinCompat(new string[] { moduleManifestPath }, null, new ImportModuleOptions());
+                        IList<PSModuleInfo> moduleProxies = ImportModulesUsingWinCompat(new string[] { moduleManifestPath }, null, options);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs]: Why do we need the change?

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs (hunk 1 above)
Comment (anmenaga) [reply src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs]: This is the fix for item 2. This scenario is covered by test `-NoClobber and -Scope work with implicit WinCompat`.

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 [hunk 2]
```diff
@@ -538,6 +553,63 @@ Describe "Additional tests for Import-Module with WinCompat" -Tag "Feature" {
...
+
+        AfterAll {
+            Restore-ModulePath
+        }
+
+        AfterEach {
+            Get-Module $allModules | Remove-Module -Force
+        }
+
+        It 'WinCompat process does not inherit PowerShell-Core-specific paths' {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: Looks like other test cases in this file explicit skip non-Windows, should you do this here?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 (hunk 2 above)
Comment (anmenaga) [reply test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: For these new tests this is already done in their `Describe -> BeforeAll`.
However the last test that was added had copy-paste leftover in a form of the test-level `-Skip`, so I cleaned that.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by codykonior** [comment]
The Get-PSSession fix is critical for anyone who wants to run this in critical production environment and so uses $ErrorActionPreference = 'Stop'. 

It cannot be overridden manually with -ErrorActionPreference:Continue; the only workaround is to set the variable to Continue and then run it, but then you require extra code to detect and parse if any errors occurred loading the module.

Hope this gets merged in and made available ASAP.

**Comment by anmenaga** [comment]
@iSazonov looks like this is ready for merge.

**Pull request merged**